ui-common/src/components/ServiceMapModal/ServiceMapModal.tsx

```diff
@@ -137,15 +138,16 @@ export const ServiceMapModal: React.FC<ServiceMapModalProps> = ({ isOpen, onClos
         setFilteredServices(filterService)
         const newGraphData: GraphData = {
             nodes: serviceMapApiData.nodes?.map(mapNodesDatatoGraph).filter(node => filterService.includes(node.label)),
-            edges: serviceMapApiData.edges?.filter(edge => filterProt.includes(edge.protocol.name)).map(mapEdgesDatatoGraph)
+            edges: serviceMapApiData.edges?.filter(edge => filterProt.includes(edge.protocol.abbr)).map(mapEdgesDatatoGraph)
         }
         setGraphData(newGraphData);
     }
 
     useEffect(() => {
-        const resolvedServices = getServicesForFilter.map(x => x.key).filter(serviceName => !Utils.isIpAddress(serviceName))
-        setFilteredServices(resolvedServices)
-        filterServiceMap(filteredProtocols, resolvedServices)
+        let mergeWithPrev = getServicesForFilter.map(x => x.key).filter(serviceName => !Utils.isIpAddress(serviceName))
+        if (filteredServices.length > 0)
```

Comment:
add comment: // only after refresh

Reply:
if (filteredServices.length > 0) return

Reply:
raise this if statement up to the top of useEffect and change the filterServiceMap func call to:

filterServiceMap(filteredProtocols, getServicesForFilter.map(x => x.key).filter(serviceName => !Utils.isIpAddress(serviceName)))